misc/python/materialize/cli/ci_flake_detect.py

```diff
@@ -0,0 +1,116 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# ci_flake_detect.py - Detect flaky test failures based on comparing logs to patterns from Github issues
+
+import argparse
+import re
+import sys
+from typing import Any, List
+
+import junit_xml
+import requests
+
+from materialize import ci_util
+
+CI_RE = re.compile("^ci-regexp: (.*)$", flags=re.MULTILINE)
+FAILURE_RE = re.compile("(panicked|Out of memory)")
```

Comment:
Lets make this regular expression match all the examples in the ticket:
- the casing of "Out of memory" is different if the process being killed is inside a memory-restricted container or not :-( .  
- include "internal error" and "FATAL" for the CRDB errors